### ISSUE TITLE
Support interop in tests with legacy cairo contracts

### ISSUE THREAD

**Issue by Arcticae** [software-mansion/protostar#1579]
We should allow user to declare and deploy both old and new contracts in tests.
We should have separate cheatcodes for both.

**Comment by MaksymilianDemitraszek** [comment]
We probably want to add `declare_legacy` cheatcode

**Comment by Arcticae** [comment]
Blocked by #1526

**Comment by MaksymilianDemitraszek** [comment]
closing in favor of #1617
